### PR TITLE
Add distillation component

### PR DESCRIPTION
**Pull request by novaturient95**
Add components for distillation 

validation run: [test](https://ml.azure.com/experiments/id/5f2850e5-924c-49d6-9d34-0f1b0d0d9413/runs/crimson_rhythm_0bt37t7789?wsid=%2Fsubscriptions%2F75703df0-38f9-4e2e-8328-45f6fc810286%2FresourceGroups%2Frg-sasumai%2Fproviders%2FMicrosoft.MachineLearningServices%2Fworkspaces%2Fsasum-westus3-ws&tid=7f292395-a08f-4cc0-b3d0-a400b023b0d2#)

### CODE REVIEW

**Review by skasturi — DISMISSED** [review]

File: assets/training/distillation/components/data_generation/src/common/constants.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Data generatior constants."""
+
+
+# COMPONENT META
+COMPONENT_NAME = "oss_generate_data"
```
Comment (skasturi) [inline assets/training/distillation/components/data_generation/src/common/constants.py:8]: This needs a change

File: assets/training/distillation/components/data_generation/asset.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+type: component
+spec: spec.yaml
+categories: ["Foundational Models", "Finetune"]
```
Comment (skasturi) [inline assets/training/distillation/components/data_generation/asset.yaml:3]: May be add distillation also as a category

File: assets/training/distillation/components/data_generation/src/common/utils.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+from azure.ai.ml.identity import AzureMLOnBehalfOfCredential
+from azure.identity import AzureCliCredential, ManagedIdentityCredential
+from azureml.acft.common_components import get_logger_app
+from azureml.core import Run, Workspace
+from azureml.core.run import _OfflineRun
+
+from typing import Union
+
+
+logger = get_logger_app("azureml.acft.contrib.hf.nlp.entry_point.data_import.data_import")
```
Comment (skasturi) [inline assets/training/distillation/components/data_generation/src/common/utils.py:19]: We might want to change this as well? 

Also, I am seeing logs being scrubbed. How do we make sure they are pushed out?

File: assets/training/distillation/components/data_generation/src/generate_data.py [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+    get_workspace_mlclient,
+    get_online_endpoint_key,
+    get_online_endpoint_url,
+    get_serverless_endpoint_key,
+    get_serverless_endpoint_url,
+    retry,
+)
+
+
+logger = get_logger_app("azureml.acft.contrib.hf.nlp.entry_point.data_import.data_import")
```
Comment (skasturi) [inline assets/training/distillation/components/data_generation/src/generate_data.py:58]: Does this scrub the logs or allow to pass through?

File: assets/training/distillation/components/data_generation/asset.yaml (hunk 2 above)
Comment (novaturient95) [reply assets/training/distillation/components/data_generation/asset.yaml:3]: Done

File: assets/training/distillation/components/data_generation/src/generate_data.py (hunk 4 above)
Comment (novaturient95) [reply assets/training/distillation/components/data_generation/src/generate_data.py:58]: we do see logs. In appinsights these logs would be filtered out: 

image

Comment (novaturient95) [reply assets/training/distillation/components/data_generation/src/generate_data.py:58]: ex: 

image

**Review by skasturi — APPROVED** [review]

**Pull request merged**